src/image/imageloadobserver.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module image/image/imageloadobserver
+ */
+
+import DomEventObserver from '@ckeditor/ckeditor5-engine/src/view/observer/domeventobserver';
+import DOMEmitterMixin from '@ckeditor/ckeditor5-utils/src/dom/emittermixin';
+
+/**
+ * Observes all images added to the {@link module:engine/view/document~Document},
+ * fires {@link module:engine/view/document~Document#event:imageLoad}` event and
+ * {@link module:engine/view/view~View#render renders} the View every time when new image
+ * has been loaded.
+ *
+ * **Note:** This event is not fired for images that has been added to the document as `complete` (already loaded).
+ *
+ * @extends module:engine/view/observer/domeventobserver~DomEventObserver
+ */
+export default class ImageLoadObserver extends DomEventObserver {
+	constructor( view ) {
+		super( view );
+
+		/**
+		 * List of img DOM elements that are observed by this observer.
+		 *
+		 * @private
+		 * @type {Set.<HTMLElement>}
+		 */
+		this._observedElements = new Set();
```

Comment:
It's better if this is `WeakSet`. Or that you clear it up in the `destroy()`.

Reply:
Not sure if we can do both. You can not iterate over `WeakSet`.